axelrod/tests/unit/test_fingerprint.py

```diff
@@ -1,6 +1,11 @@
 import unittest
```

Comment:
Reviewing this diff is going to take a bit of work because it looks like a big rewrite (chunks of addition/deletion). If you could write a comment with some pointers that would be helpful.